### PR TITLE
Reduce-References-RBVariableNode-subclasses2

### PR DESCRIPTION
**Pull request by MarcusDenker**
another pass to reduce the users of RBVariableNode references. Instead, use RBVariableNode. The name analysis takes care of the rest 

There are some left that will be done in following PRs:

RBSelfNode
		- NewTools: testShouldConsiderHaltNode
		- identifierNamed:at: 
		
RBSuperNode
		- identifierNamed:at: 
		- rewriteASTToSimulateExecutionInADifferentContext:
		- testDoesApplyForSuperMessageSendNodes
		
RBThisContextNode
		identifierNamed:at: 
		testThisContextNodeDump -> needs visitor
		testBuildThisContextHeuristic -> needs visitor

### CODE REVIEW

File: src/AST-Core/RBSequenceNode.class.st [hunk 1]
```diff
@@ -176,7 +176,7 @@ RBSequenceNode >> addSelfReturn [
 	| node |
 	self lastIsReturn
 		ifTrue: [ ^ self statements last ].
-	node := RBReturnNode value: (RBSelfNode new).
+	node := RBReturnNode value: (RBVariableNode named: 'self').
```
Comment (dionisiydk) [inline src/AST-Core/RBSequenceNode.class.st]: We should avoid such strings ('self', 'super'). Pablo introduced something like "RBVariableNode selfNode" in his PR. We need to push it here.

File: src/OpalCompiler-Core/RBMethodNode.extension.st [hunk 2]
```diff
@@ -38,12 +38,12 @@ RBMethodNode >> ensureCachedArgumentNames [
 RBMethodNode class >> errorMethodNode: selector errorMessage: messageText [	
 	| message |
 	message := RBMessageNode 
-		receiver: RBSelfNode new
+		receiver: (RBVariableNode named: self)
```
Comment (dionisiydk) [inline src/OpalCompiler-Core/RBMethodNode.extension.st]: and here is a bug due to 'self' duplications.

**Review by dionisiydk — CHANGES_REQUESTED** [review]

File: src/AST-Core/RBSequenceNode.class.st (hunk 1 above)
Comment (MarcusDenker) [reply src/AST-Core/RBSequenceNode.class.st]: My idea was to make sure that people understand that Variables have no meaning before name analysis. We need to be consistent: We can not do it for temps (humanly impossible to know what kind of temp and which offset). Sp why do it for Self?

File: src/OpalCompiler-Core/RBMethodNode.extension.st (hunk 2 above)
Comment (MarcusDenker) [reply src/OpalCompiler-Core/RBMethodNode.extension.st]: fixed

File: src/AST-Core/RBSequenceNode.class.st (hunk 1 above)
Comment (MarcusDenker) [reply src/AST-Core/RBSequenceNode.class.st]: Hmmm "RBVariableNode selfNode" makes sense... but only for self, super and thisContext.

Comment (dionisiydk) [reply src/AST-Core/RBSequenceNode.class.st]: Just that there is a duplication of code now. 
But this particular method has already semantics meaning: we return a self. So it needs to build a valid structure.

Comment (MarcusDenker) [reply src/AST-Core/RBSequenceNode.class.st]: I will update the OR to use #selfNode, superNode and thisContextNode

**Review by dionisiydk — APPROVED** [review]

**Pull request merged**